metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/aop/ReleaseDateAspect.java

```diff
@@ -69,12 +68,16 @@ public Object filterOnReleaseDateAdviceFindAll(ProceedingJoinPoint proceedingJoi
     @Around("execution(* org.springframework.data.repository.CrudRepository.findOne(java.io.Serializable))")
     public Object filterOnReleaseDateAdviceFindOne(ProceedingJoinPoint proceedingJoinPoint) throws Throwable {
         Object result = proceedingJoinPoint.proceed();
+        return processReleaseControlledObject(result);
+    }
 
-        if ( result == null ||
-                (result instanceof Study && ((Study) result).getReleaseDate().isAfter(LocalDate.now())) ) {
-            return null;
+    private Object processReleaseControlledObject(Object result) {
```

Comment:
I would suggest using a name that states more clearly the functionality of the method, such as `getObjectUnlessBeforeReleaseDate`.

Reply:
Agreed, fixed in https://github.com/EBIvariation/metadata-ws/pull/53/commits/b12ad94aa306c0e8eb84f967c15b5cb6baa5fc2c